templates/fine-tune-llm_v2/end-to-end-examples/fine-tune-preference/README.ipynb

```diff
@@ -0,0 +1,1028 @@
+      "an 'exceptional' education for any child and that there is no need to pay fees\n",
+      "for schooling. The deadline for applying to Ark Putney for 2013-14 was last\n",
+      "October and the school is part of the Ark academy chain, set up in 2004, whose\n",
+      "chairman is Paul Marshall, one of the Liberal Democrats' biggest donors.\n",
+      "\n",
+      "REJECTED RESPONSE:\n",
+      "Deputy Prime Minister Nick Clegg has defended the possibility of sending his\n",
+      "children to private schools, stating that their education should not be used as\n",
+      "a political football. He and his wife Miriam have not visited their local state\n",
+      "school, but the headmaster of Ark Putney academy in London, where they were\n",
+      "invited to attend, said the school could provide an \"exceptional\" education for\n",
+      "any child and that parents should come to see it firsthand before making a\n",
+      "decision.\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "for row in example_rows:\n",
+    "    print_wrapped(\"PROMPT\", row[\"chosen\"][0][\"content\"])\n",
+    "    print_wrapped(\"CHOSEN RESPONSE\", row[\"chosen\"][1][\"content\"])\n",
+    "    print_wrapped(\"REJECTED RESPONSE\", row[\"rejected\"][1][\"content\"])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Step 2: Fine-tuning\n",
+    "\n",
+    "Now that we have the pre-processed dataset, we are ready to fine-tune `Mistral-7B-Instruct-v0.1` using DPO. On Anyscale, we've created an easy-to-use interface to do preference-tuning using DPO. We leverage Ray to overlap reference model log-probability calculation with model training to improve GPU utilization. Most implementations compute log probabilities synchronously with model training,\n",
+    "\n",
+    "![hf model](./assets/hf_dpo.png)\n",
+    "\n",
+    "While our implementation using Ray is asynchronous:  \n",
+    "\n",
+    "\n",
+    "![assistant model](./assets/anyscale_dpo.png)\n",
+    "\n",
+    "Further, our use of Ray Data also implies that the compute configuration for the reference model can be completely decoupled with the policy model. For example, reference model calculation can run on a different node (with configurable number of GPUs, etc) with zero code changes needed. \n",
+    "\n",
+    "> **NOTE** Make sure you've gove over the [user guides](https://docs.anyscale.com/category/fine-tuning-beta) for fine-tuning to understand the different configurations available\n",
+    "\n",
+    "To get started with DPO training, we provide the config for DPO in [configs/mistral_dpo_summarization.yaml](configs/mistral_dpo_summarization.yaml) . You can add your `WANDB_API_KEY` as an environment variable in the dependencies tab if you wish to track progress of your run on WandB.\n",
+    "\n",
+    "\n",
+    " 🔄 REPLACE the training and validation file paths in the config with the output file paths in the previous step"
+   ]
+  },
+  {
```

Comment:
Please lighten the content of this YAML. only include things that you are explicitly overriding its default. for example output_dir, use_flash_attention_2 etc we should remove those. 

Another thing: This shows LoRA, are we doing LoRA or full param? Let's be careful about it and provide both if possible.

Reply:
Yeah sure I'll add a note that they can tweak this. Evaluation already provides and mentions this flexibility.